components/layout_thread/lib.rs

```diff
@@ -1664,35 +1668,46 @@ impl ProfilerMetadataFactory for LayoutThread {
     }
 }
 
-// The default computed value for background-color is transparent (see
-// http://dev.w3.org/csswg/css-backgrounds/#background-color). However, we
-// need to propagate the background color from the root HTML/Body
-// element (http://dev.w3.org/csswg/css-backgrounds/#special-backgrounds) if
-// it is non-transparent. The phrase in the spec "If the canvas background
-// is not opaque, what shows through is UA-dependent." is handled by rust-layers
-// clearing the frame buffer to white. This ensures that setting a background
-// color on an iframe element, while the iframe content itself has a default
-// transparent background color is handled correctly.
-fn get_root_flow_background_color(flow: &mut Flow) -> webrender_api::ColorF {
-    let transparent = webrender_api::ColorF { r: 0.0, g: 0.0, b: 0.0, a: 0.0 };
-    if !flow.is_block_like() {
-        return transparent;
+/// Propagate background if necessary from the <body> to the document root.
+///
+/// Backgrounds by default are transparent. This ensures that iframes
+/// are displayed with a background from the surrounding context.
+///
+/// Spec: https://drafts.csswg.org/css-backgrounds/#body-background
+fn determine_root_background(root_flow: &mut Flow) {
+    if !root_flow.is_block_like() {
+        return;
+    }
+    let root_block = root_flow.as_mut_block();
+    let root_style = ServoArc::make_mut(&mut root_block.fragment.style);
```

Comment:
It looks unfortunate to mutate the style from here... We should at least move the `make_mut` to where we know we are going to change it.

Reply:
> It looks unfortunate to mutate the style from here

Where should it be done instead?